lib/core/state/mongo-server.js

```diff
@@ -0,0 +1,51 @@
+const ReplicationCommands = require('../commands/replication-commands')
+
+// Direct mappings to https://docs.mongodb.com/manual/reference/replica-states/
+const SERVER_REPLICATION_STATUS = {
+  UNINITIALIZED: -1, // custom hydra state
+  STARTUP: 0,
+  PRIMARY: 1,
+  SECONDARY: 2,
+  RECOVERING: 3,
+  STARTUP2: 5,
+  UNKNOWN: 6,
+  ARBITER: 7,
+  DOWN: 8,
+  ROLLBACK: 9
+}
+
+class MongoServerState {
+  constructor(hostname, client) {
```

Comment:
@drubin What do you think of making this `constructor(client, settings)` so we can extend this in the future by for example settings like `delayedMember`, `priority`, ....

Reply:
Hmm, but thats rather a ReplicationConfiguration thing. There must be a good way to combine these.

Reply:
How about something like this:

```
class ReplicationConfiguration {
  constructor({
    replicaSetName,
    servers, // array of ServerReplicationConfiguration
    configServer = false
  }) {
    this.replicaSetName = replicaSetName
    this.servers = servers
    this.configServer = configServer
  }

class ServerReplicationConfiguration {
constructor(settings, server: MongoServerState) {
this.priority = settings.priority || 1
}
}
```

Reply:
I  am a fan of this.